preprocessing/analyze_chars.py

```diff
@@ -1,80 +1,88 @@
-# Use a screenshot of all ASCII chars to determine the luminance value of each of them
+"""Use a screenshot of all ASCII chars to determine the
+luminance value of each of them."""
 
 import os
 import numpy as np
 import json
 from math import floor
 from PIL import Image
+from functools import partial
+
+
+# Size of each characters in the screenshot
+CHAR_WIDTH = 11
+CHAR_HEIGHT = 23
+
+def closest_lookup(dictionary, key):
+	"""Will find the value for the key which is closest to the key given.
+
+	The dictionary keys should all be numbers."""
+	return dictionary[min(dictionary.keys(), key=lambda x:abs(x - key))]
+
+
+def coordinates_of_char(char_code):
+	"""Given a char_code, will return the coordinates of the char in chars.png"""
+	i = char_code - 33
+	x = (i % 8) * CHAR_WIDTH
+	y = (i // 8) * CHAR_HEIGHT
+	return x, y
 
 def average_val(pixels, y, x, y_step, x_step):
-    # Return average value in boolean pixels numpy array in the given rectangle
-    # Expects rectangle coordinates to be in the range of the pixels array
+    """Return average value in boolean pixels numpy array in the given rectangle
+    Expects rectangle coordinates to be in the range of the pixels array."""
     if y_step == 0:
         raise Exception('y_step must be non-zero')
     if x_step == 0:
         raise Exception('x_step must be non-zero')
 
-    start_y = int(floor(y))
-    end_y = int(floor(y+y_step))
-    start_x = int(floor(x))
```

Comment:
int(something_which_is_an_integer_already) does nothing but add noise and confusion

Reply:
these variables were all floats (completely undocumented to be fair), so this did make sense at the time. it's kinda confusing though, I agree, so no more floats being passed around

Reply:
ah, this is in analyze_chars, so they are ints. I just copied this function from ascii.py (where it is called with floats) for convenience. This file is more of a one-off hacky script just to produce the luminosity-to-char JSON, not much care went into it